docs/user-guide/kubectl-cheatsheet.md

```diff
@@ -0,0 +1,119 @@
+
+$ kubectl create -f <dir>               # create resources in all .json, .yml, and .yaml files in dir
+
+# Create from a URL
+$ kubectl create -f http://www.fpaste.org/279276/48569091/raw/
+
+# Create multiple YAML objects from stdin
+$ cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox-sleep
+spec:
+  containers:
+  - name: busybox
+    image: busybox
+    args:
+    - sleep
+    - "1000000"
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox-sleep-less
+spec:
+  containers:
+  - name: busybox
+    image: busybox
+    args:
+    - sleep
+    - "1000"
+EOF
+
+# Create a secret with several files
+$ cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+data:
+  password: $(echo "s33msi4" | base64)
+  username: $(echo "jane" | base64)
+EOF
+
+# TODO: kubectl-explain example
+```
+
+
+## Getting, Viewing, Finding Resources
+
+```console
+# Columnar output
+$ kubectl get services                          # List all services in the namespace
+$ kubectl get pods --all-namespaces             # List all pods in all namespaces
+$ kubectl get pods -o wide                      # List all pods in the namespace, with more details
+$ kubectl get replicationcontroller <rc-name>   # List a particular RC
+
+# Verbose output
+$ kubectl describe nodes <node-name>
+$ kubectl describe pods <pod-name>
+$ kubectl describe pods/<pod-name>              # Equivalent to previous
+$ kubectl describe pods <rc-name>               # Lists pods created by <rc-name> using common prefix
+
+# List Services Sorted by Name
+$ kubectl get services --sort-by=.metadata.name
+
+## List pods Sorted by Restart Count
+$ kubectl get pods --sort-by=.status.containerStatuses[0].restartCount
+
+# Get the version label of all pods with label app=cassandra
+$ kubectl get pods --selector=app=cassandra rc -o 'jsonpath={.items[*].metadata.labels.version}'
+
+# Get ExternalIPs of all nodes
+$ kubectl get nodes -o jsonpath='{.items[*].status.addresses[?(@.type=ExternalIP)].address}'
+
+# List Names of Pods that belong to Particular RC
+$ sel=$(kubectl get rc <rcname> --output=jsonpath={.spec.selector})`
+$ pods=$(kubectl get pods --selector=$sel --output=jsonpath={.items..metadata.name})`
+
+# Check which nodes are ready
+$ kubectl get nodes -o jsonpath='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'| tr ';' "\n"  | grep "Ready=True" 
+```
+
+## Modifying and Deleteing Resources
+
+```console
+$ kubectl label pods <pod-name> new-label=awesome                  # Add a Label
+$ kubectl annotate pods <pod-name> icon-url=http://goo.gl/XXBTWq   # Add an annotation
+
+# TODO: examples of kubectl edit, patch, delete, replace, scale, and rolling-update commands.
+```
+
+## Interacting with running Pods
+
+```console
+$ kubectl logs <pod-name>         # dump pod logs (stdout)
+$ kubectl logs -f <pod-name>      # stream pod logs (stdout) for a while
```

Comment:
I was intentionally vague because I could not determine what the status was of https://github.com/kubernetes/kubernetes/issues/9013, nor if we might reverse whatever that status was in the future.